### DISCUSSION TITLE
Sign is unnecessarily async

### DISCUSSION THREAD

**Discussion by paulmillr** [wevm/viem#1408] [Idea / Feature Request]
This includes all methods that call `sign` - there are many of them. They could be sync, because underlying primitive is sync.

This is breaking change.

https://github.com/wagmi-dev/viem/blob/d5eb4936438950714bef63c48b71d1d21358af60/src/accounts/utils/sign.ts#L20-L24

**Comment by jxom** [comment]
Will add a note to change this to sync for v2.

Background: it was originally async, and we didn’t want to break users (yet) so we left it async.

**Comment by paulmillr** [comment]
this hasn't been fixed in v2

**Reply by tmm** [reply]
Forgot to add a TODO comment for v2 😬 Coming in [v3](https://github.com/wevm/viem/blob/main/src/accounts/utils/sign.ts#L1)
